src/libsync/syncengine.cpp

```diff
@@ -889,7 +916,7 @@ void SyncEngine::slotDiscoveryJobFinished(int discoveryResult)
     _journal->commit("post treewalk");
 
     _propagator = QSharedPointer<OwncloudPropagator>(
-        new OwncloudPropagator (_account, _localPath, _remoteUrl, _remotePath, _journal));
+        new OwncloudPropagator (_account, _localPath, _remoteUrl.path(), _remotePath, _journal));
```

Comment:
I think changing the remoteUrl to QUrl is a dangerous thing. We need to make sure that _remoteUrl.path() still contains the same as the previous string (trailing slashes and stuff...) What benefit does this change bring?